NewHaven/src/Player.cpp

```diff
@@ -2,14 +2,14 @@
 #include "GBMap.h"
 Player::Player(){
     village = new VGMap();
-    resource_score = new ResourceScore();
+    resource_score = new ResourceTracker();
     vb_score = new ScoreCalculator();
     my_hand = new Hand();
 }
 
 Player::Player(const Player &player) {
     village = new VGMap(*player.village);
-    resource_score = new ResourceScore(*player.resource_score);
+    //resource_score = new ResourceScore(*player.resource_score);
     vb_score = new ScoreCalculator(*player.vb_score);
     my_hand = new Hand(*player.my_hand);
```

Comment:
You don't copy the resourceTracker and you also don't initialize it. You need to pick one or the other.
Also there's unnecessary copying. Using the copy constructor and the copy assignment at the same time isn't efficient.

Reply:
that is my bad i had commented it out for testing and completely forgot to add one for ResourceTracker ooops lol